flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/FlinkUtils.java

```diff
@@ -61,6 +61,18 @@ public static DefaultConfig loadDefaultConfig() {
         return new DefaultConfig(operatorConfig, flinkConfig);
     }
 
+    public static Configuration getFlinkEffectiveConfig(FlinkDeployment flinkApp) {
+        try {
+            final Configuration effectiveConfig =
+                    FlinkConfigBuilder.buildFlinkFrom(
+                            flinkApp, loadConfiguration(EnvUtils.get(EnvUtils.ENV_FLINK_CONF_DIR)));
```

Comment:
We should avoid loading the configuration again and again. Best would be to only do it once here or in the validtor